### DISCUSSION TITLE
Unexpected CIE XYZ output for emission spectra with "colour.colorimetry.sd_to_XYZ_integration" definition.

### DISCUSSION THREAD

**Discussion by heinemannj** [colour-science/colour#849] [Q&A] [answered]
By the calculation of CIE XYZ from an emission spectrum of a white Nichia LED (NSPW500GS-K1), measured with a selfmade webcam spectrometer, I'm facing the following strange results for XYZ:

```
---------------------------------------------------------------------------------------------------------------------

SpectralDistribution([[  200.,     0.],
                      [  201.,     0.],
                      [  202.,     0.],
       ..., 
                      [ 1198.,     0.],
                      [ 1199.,     0.],
                      [ 1200.,     0.]],
                     interpolator=SpragueInterpolator,
                     interpolator_kwargs={},
                     extrapolator=Extrapolator,
                     extrapolator_kwargs={'method': 'Constant', 'left': None, 'right': None})


SpectralShape(200.0, 1200.0, 1.0)
Spectral distribution uniformity:  True

---------------------------------------------------------------------------------------------------------------------

CIE 1931 2 Degree Standard Observer

XYZ_ColourMatchingFunctions([[  3.60000000e+02,   1.29900000e-04,   3.91700000e-06,
          6.06100000e-04],
                             [  3.61000000e+02,   1.45847000e-04,   4.39358100e-06,
          6.80879200e-04],
                             [  3.62000000e+02,   1.63802100e-04,   4.92960400e-06,
          7.65145600e-04],
       ..., 
                             [  8.28000000e+02,   1.43944000e-06,   5.19808000e-07,
          0.00000000e+00],
                             [  8.29000000e+02,   1.34197700e-06,   4.84612000e-07,
          0.00000000e+00],
                             [  8.30000000e+02,   1.25114100e-06,   4.51810000e-07,
          0.00000000e+00]],
                            labels=['x_bar', 'y_bar', 'z_bar'],
                            interpolator=SpragueInterpolator,
                            interpolator_kwargs={},
                            extrapolator=Extrapolator,
                            extrapolator_kwargs={'method': 'Constant', 'left': None, 'right': None})

---------------------------------------------------------------------------------------------------------------------

Spectral distribution data                                               :  Emissive Case
cmfs                                                                     :  CIE 1931 2 Degree Standard Observer
illuminant (LED spectral data)                                           :  5000K-Nichia-NSPW500GS-K1
k                                                                        :  None
XYZ by sd_to_XYZ(sd, cmfs, illuminant, k, method='Integration')          :  [ 81.43209916  80.37617556  76.84025725]
XYZ by colour.colorimetry.sd_to_XYZ_integration(sd, cmfs, illuminant, k) :  [ 81.43209916  80.37617556  76.84025725]
XYZ by self-made integration                                             :  [82.440269498908407, 82.854328696303156, 78.687968773869486]

---------------------------------------------------------------------------------------------------------------------
```

Complete code is also attached:
```
print('---------------------------------------------------------------------------------------------------------------------\n')
# Spectral distribution data - Emissive Case
sd = SpectralDistribution(data, name=name)
print(repr(sd))
print('\n')
print(repr(sd.shape))
print("Spectral distribution uniformity: ", sd.is_uniform())

cmfs = MSDS_CMFS['CIE 1931 2 Degree Standard Observer']
print('\n---------------------------------------------------------------------------------------------------------------------\n')
print('CIE 1931 2 Degree Standard Observer\n')
print(repr(cmfs))
illuminant = SpectralDistribution(data)
# illuminant = SDS_ILLUMINANTS['E']
k=None;
# k = 1;

print('\n---------------------------------------------------------------------------------------------------------------------\n')
# Calculation of *CIE XYZ* tristimulus values - Emissive Case
XYZ = sd_to_XYZ(sd, cmfs, illuminant, k, method='Integration')
XYZ_Integration = colour.colorimetry.sd_to_XYZ_integration(sd, cmfs, illuminant, k)
#XYZ_Integration = colour.colorimetry.sd_to_XYZ_integration(sd)

XYZ_Check=[0,0,0]
wavelength = 360
while wavelength != 831:
    XYZ_Check[0]+=cmfs[wavelength][0]*sd[wavelength]
    XYZ_Check[1]+=cmfs[wavelength][1]*sd[wavelength]
    XYZ_Check[2]+=cmfs[wavelength][2]*sd[wavelength]
    wavelength += 1

print("Spectral distribution data                                               :  Emissive Case")
print("cmfs                                                                     :  CIE 1931 2 Degree Standard Observer")
print("illuminant (LED spectral data)                                           : ", name)
print("k                                                                        : ", k)
print("XYZ by sd_to_XYZ(sd, cmfs, illuminant, k, method='Integration')          : ",XYZ)
print("XYZ by colour.colorimetry.sd_to_XYZ_integration(sd, cmfs, illuminant, k) : ",XYZ_Integration)
print("XYZ by self-made integration                                             : ",XYZ_Check)
print('\n---------------------------------------------------------------------------------------------------------------------')
```
I'm a bloddy beginner in spectrometry and colorimetry.
Any help is appreciated.

[XYZ.pyw.txt](https://github.com/colour-science/colour/files/6988152/XYZ.pyw.txt)

**Comment by thomasmansencal** [comment]
Hi @heinemannj,

We are generally using **ASTM E308-15** practise and according to it, the spectral shape for computation is as follows: `SPECTRAL_SHAPE_DEFAULT = SpectralShape(360, 780, 1)`. Can you check your numbers by reducing the range of your computations? Given you a higher CIE XYZ output, it seems to be related.

Cheers,

Thomas

**Reply by heinemannj** [reply]
Hi,
the emission spectum of the LED is in the range of **398 - 741** nm:
![image](https://user-images.githubusercontent.com/3251642/129498772-c29f7311-2628-4186-b951-a2e66092972d.png)
Reducing the range for XYZ integration to 360-780 nm is providing the same figures:

```
Spectral distribution data                                               :  Emissive Case
cmfs                                                                     :  CIE 1931 2 Degree Standard Observer
illuminant (LED spectral data)                                           :  5000K-Nichia-NSPW500GS-K1
k                                                                        :  None
XYZ by sd_to_XYZ(sd, cmfs, illuminant, k, method='Integration')          :  [ 81.43209916  80.37617556  76.84025725]
XYZ by colour.colorimetry.sd_to_XYZ_integration(sd, cmfs, illuminant, k) :  [ 81.43209916  80.37617556  76.84025725]
XYZ by self-made integration                                             :  [82.440269498908407, 82.854328696303156, 78.687968773869486]
```

Cheers
Joerg

**Reply by thomasmansencal** [reply]
Hi Joerg,

I just realised that `k` is set to `None` thus the output is normalised by the automatically computed value for `k`:

```python
k = 100 / (np.sum(y_bar * S) * dw) if k is None else k
```

Try to set `k=1` and hopefully it should match!

**Comment by heinemannj** [comment]
Already tested yesterday - k=1 provides XYZ vales far away from my self made XYZ integration:

```
# Calculation of *CIE XYZ* tristimulus values - Emissive Case
XYZ = sd_to_XYZ(sd, cmfs, illuminant, k=1, method='Integration')
XYZ_Integration = colour.colorimetry.sd_to_XYZ_integration(sd, cmfs, illuminant, k=1)
```

```
Spectral distribution data                                               :  Emissive Case
cmfs                                                                     :  CIE 1931 2 Degree Standard Observer
illuminant (LED spectral data)                                           :  5000K-Nichia-NSPW500GS-K1
k                                                                        :  1
XYZ by sd_to_XYZ(sd, cmfs, illuminant, k, method='Integration')          :  [ 67.47001911  66.59514069  63.66547932]
XYZ by colour.colorimetry.sd_to_XYZ_integration(sd, cmfs, illuminant, k) :  [ 67.47001911  66.59514069  63.66547932]
XYZ by self-made integration                                             :  [82.440269498908407, 82.854328696303156, 78.687968773869486]
```

Question:
Is my XYZ integration approach for an LED emmision spectrum valid?

```
XYZ_Check=[0,0,0]
wavelength = 360
while wavelength != 831:
    XYZ_Check[0]+=cmfs[wavelength][0]*sd[wavelength]
    XYZ_Check[1]+=cmfs[wavelength][1]*sd[wavelength]
    XYZ_Check[2]+=cmfs[wavelength][2]*sd[wavelength]
    wavelength += 1
```

**Reply by thomasmansencal** [reply]
It is uncommon but on the paper and assuming 1nm, I think _it should work_ but I'm overseeing something here quite obviously. Would you have the spectrum handy so that I can check?

**Reply by heinemannj** [reply]
# Spectral data - Emissive Case
name = "5000K-Nichia-NSPW500GS-K1"
data = {
200	:	0	,
201	:	0	,
202	:	0	,
203	:	0	,
204	:	0	,
205	:	0	,
206	:	0	,
207	:	0	,
208	:	0	,
209	:	0	,
210	:	0	,
211	:	0	,
212	:	0	,
213	:	0	,
214	:	0	,
215	:	0	,
216	:	0	,
217	:	0	,
218	:	0	,
219	:	0	,
220	:	0	,
221	:	0	,
222	:	0	,
223	:	0	,
224	:	0	,
225	:	0	,
226	:	0	,
227	:	0	,
228	:	0	,
229	:	0	,
230	:	0	,
231	:	0	,
232	:	0	,
233	:	0	,
234	:	0	,
235	:	0	,
236	:	0	,
237	:	0	,
238	:	0	,
239	:	0	,
240	:	0	,
241	:	0	,
242	:	0	,
243	:	0	,
244	:	0	,
245	:	0	,
246	:	0	,
247	:	0	,
248	:	0	,
249	:	0	,
250	:	0	,
251	:	0	,
252	:	0	,
253	:	0	,
254	:	0	,
255	:	0	,
256	:	0	,
257	:	0	,
258	:	0	,
259	:	0	,
260	:	0	,
261	:	0	,
262	:	0	,
263	:	0	,
264	:	0	,
265	:	0	,
266	:	0	,
267	:	0	,
268	:	0	,
269	:	0	,
270	:	0	,
271	:	0	,
272	:	0	,
273	:	0	,
274	:	0	,
275	:	0	,
276	:	0	,
277	:	0	,
278	:	0	,
279	:	0	,
280	:	0	,
281	:	0	,
282	:	0	,
283	:	0	,
284	:	0	,
285	:	0	,
286	:	0	,
287	:	0	,
288	:	0	,
289	:	0	,
290	:	0	,
291	:	0	,
292	:	0	,
293	:	0	,
294	:	0	,
295	:	0	,
296	:	0	,
297	:	0	,
298	:	0	,
299	:	0	,
300	:	0	,
301	:	0	,
302	:	0	,
303	:	0	,
304	:	0	,
305	:	0	,
306	:	0	,
307	:	0	,
308	:	0	,
309	:	0	,
310	:	0	,
311	:	0	,
312	:	0	,
313	:	0	,
314	:	0	,
315	:	0	,
316	:	0	,
317	:	0	,
318	:	0	,
319	:	0	,
320	:	0	,
321	:	0	,
322	:	0	,
323	:	0	,
324	:	0	,
325	:	0	,
326	:	0	,
327	:	0	,
328	:	0	,
329	:	0	,
330	:	0	,
331	:	0	,
332	:	0	,
333	:	0	,
334	:	0	,
335	:	0	,
336	:	0	,
337	:	0	,
338	:	0	,
339	:	0	,
340	:	0	,
341	:	0	,
342	:	0	,
343	:	0	,
344	:	0	,
345	:	0	,
346	:	0	,
347	:	0	,
348	:	0	,
349	:	0	,
350	:	0	,
351	:	0	,
352	:	0	,
353	:	0	,
354	:	0	,
355	:	0	,
356	:	0	,
357	:	0	,
358	:	0	,
359	:	0	,
360	:	0	,
361	:	0	,
362	:	0	,
363	:	0	,
364	:	0	,
365	:	0	,
366	:	0	,
367	:	0	,
368	:	0	,
369	:	0	,
370	:	0	,
371	:	0	,
372	:	0	,
373	:	0	,
374	:	0	,
375	:	0	,
376	:	0	,
377	:	0	,
378	:	0	,
379	:	0	,
380	:	0	,
381	:	0	,
382	:	0	,
383	:	0	,
384	:	0	,
385	:	0	,
386	:	0	,
387	:	0	,
388	:	0	,
389	:	0	,
390	:	0	,
391	:	0	,
392	:	0	,
393	:	0	,
394	:	0	,
395	:	0	,
396	:	0	,
397	:	0	,
398	:	0.001251914	,
399	:	0.004771276	,
400	:	0.00996918	,
401	:	0.016908766	,
402	:	0.024952544	,
403	:	0.033233041	,
404	:	0.042527691	,
405	:	0.053645557	,
406	:	0.066245642	,
407	:	0.079935686	,
408	:	0.094805246	,
409	:	0.11313974	,
410	:	0.13492206	,
411	:	0.1588005	,
412	:	0.18448071	,
413	:	0.21104675	,
414	:	0.24216469	,
415	:	0.27682692	,
416	:	0.31441972	,
417	:	0.35374011	,
418	:	0.39542147	,
419	:	0.44266395	,
420	:	0.49247325	,
421	:	0.54044385	,
422	:	0.58895316	,
423	:	0.63603518	,
424	:	0.68201537	,
425	:	0.72504768	,
426	:	0.76307118	,
427	:	0.7959539	,
428	:	0.82338374	,
429	:	0.84579357	,
430	:	0.86214411	,
431	:	0.87429892	,
432	:	0.88577185	,
433	:	0.89575344	,
434	:	0.90469588	,
435	:	0.9124811	,
436	:	0.92089503	,
437	:	0.92926945	,
438	:	0.93657642	,
439	:	0.94254478	,
440	:	0.94814102	,
441	:	0.95392336	,
442	:	0.95936069	,
443	:	0.96370957	,
444	:	0.96633532	,
445	:	0.96778166	,
446	:	0.96832276	,
447	:	0.96725521	,
448	:	0.96483153	,
449	:	0.96022525	,
450	:	0.95468818	,
451	:	0.94769551	,
452	:	0.93955198	,
453	:	0.93103116	,
454	:	0.92230963	,
455	:	0.91303206	,
456	:	0.90166753	,
457	:	0.88777554	,
458	:	0.87104116	,
459	:	0.8496651	,
460	:	0.82254524	,
461	:	0.79172934	,
462	:	0.76052092	,
463	:	0.72841869	,
464	:	0.69726069	,
465	:	0.66668197	,
466	:	0.63675674	,
467	:	0.60826125	,
468	:	0.58121003	,
469	:	0.55397191	,
470	:	0.5265442	,
471	:	0.50108294	,
472	:	0.47923947	,
473	:	0.4576046	,
474	:	0.43820256	,
475	:	0.42078658	,
476	:	0.40593918	,
477	:	0.39366687	,
478	:	0.384087	,
479	:	0.37690256	,
480	:	0.3719796	,
481	:	0.36984056	,
482	:	0.36949141	,
483	:	0.37039412	,
484	:	0.37317562	,
485	:	0.3780733	,
486	:	0.38484054	,
487	:	0.39307792	,
488	:	0.40239157	,
489	:	0.41274206	,
490	:	0.42464409	,
491	:	0.43754235	,
492	:	0.45205168	,
493	:	0.46832685	,
494	:	0.48623719	,
495	:	0.50697984	,
496	:	0.52933474	,
497	:	0.552992	,
498	:	0.57645227	,
499	:	0.59893588	,
500	:	0.62063971	,
501	:	0.64080412	,
502	:	0.66043692	,
503	:	0.67956085	,
504	:	0.69765706	,
505	:	0.71566877	,
506	:	0.73291065	,
507	:	0.74925931	,
508	:	0.76429118	,
509	:	0.77755263	,
510	:	0.78949275	,
511	:	0.79914762	,
512	:	0.80809497	,
513	:	0.81676738	,
514	:	0.82392629	,
515	:	0.83046018	,
516	:	0.8361994	,
517	:	0.84172028	,
518	:	0.84689234	,
519	:	0.85116538	,
520	:	0.85432363	,
521	:	0.85677153	,
522	:	0.85906627	,
523	:	0.86133141	,
524	:	0.86257546	,
525	:	0.86319724	,
526	:	0.86331864	,
527	:	0.86309887	,
528	:	0.86206759	,
529	:	0.86002439	,
530	:	0.85715087	,
531	:	0.85375757	,
532	:	0.84968934	,
533	:	0.84486337	,
534	:	0.8388269	,
535	:	0.83242924	,
536	:	0.82587673	,
537	:	0.81890815	,
538	:	0.81163311	,
539	:	0.80417235	,
540	:	0.79672351	,
541	:	0.78938419	,
542	:	0.78179236	,
543	:	0.77402612	,
544	:	0.76615169	,
545	:	0.7586409	,
546	:	0.75174575	,
547	:	0.74475776	,
548	:	0.73724849	,
549	:	0.72962105	,
550	:	0.72227891	,
551	:	0.71536907	,
552	:	0.70896371	,
553	:	0.70339876	,
554	:	0.69866843	,
555	:	0.69533043	,
556	:	0.69287825	,
557	:	0.69018146	,
558	:	0.68700792	,
559	:	0.68361728	,
560	:	0.67964795	,
561	:	0.67444639	,
562	:	0.66957038	,
563	:	0.66651475	,
564	:	0.66626222	,
565	:	0.66934023	,
566	:	0.67611318	,
567	:	0.68665328	,
568	:	0.70078389	,
569	:	0.71805557	,
570	:	0.73621891	,
571	:	0.75494914	,
572	:	0.77504363	,
573	:	0.79627438	,
574	:	0.81733851	,
575	:	0.83795923	,
576	:	0.85779472	,
577	:	0.87657366	,
578	:	0.89408955	,
579	:	0.91137396	,
580	:	0.92701574	,
581	:	0.94097335	,
582	:	0.95285172	,
583	:	0.9627172	,
584	:	0.96985325	,
585	:	0.9755169	,
586	:	0.98003686	,
587	:	0.98287626	,
588	:	0.9853354	,
589	:	0.98863469	,
590	:	0.99257047	,
591	:	0.99626331	,
592	:	0.99810978	,
593	:	0.9997051	,
594	:	0.99870194	,
595	:	0.99598576	,
596	:	0.99175688	,
597	:	0.98448365	,
598	:	0.97584145	,
599	:	0.96667045	,
600	:	0.95734937	,
601	:	0.94738371	,
602	:	0.93664155	,
603	:	0.92728009	,
604	:	0.91622648	,
605	:	0.90365999	,
606	:	0.89113403	,
607	:	0.87753238	,
608	:	0.8638995	,
609	:	0.84879414	,
610	:	0.83324886	,
611	:	0.81845488	,
612	:	0.80450725	,
613	:	0.79217145	,
614	:	0.77972891	,
615	:	0.76610397	,
616	:	0.75420152	,
617	:	0.74270581	,
618	:	0.73038651	,
619	:	0.71707612	,
620	:	0.70457816	,
621	:	0.69303851	,
622	:	0.68245245	,
623	:	0.67223399	,
624	:	0.66229838	,
625	:	0.65138806	,
626	:	0.64230074	,
627	:	0.63355384	,
628	:	0.62470622	,
629	:	0.6160057	,
630	:	0.60726564	,
631	:	0.5980817	,
632	:	0.58872302	,
633	:	0.57897716	,
634	:	0.56892209	,
635	:	0.55784695	,
636	:	0.54783767	,
637	:	0.53862067	,
638	:	0.52993858	,
639	:	0.5217265	,
640	:	0.51308823	,
641	:	0.50487353	,
642	:	0.49754771	,
643	:	0.49058806	,
644	:	0.48409551	,
645	:	0.47780038	,
646	:	0.47181414	,
647	:	0.46625564	,
648	:	0.46062564	,
649	:	0.45473554	,
650	:	0.44790367	,
651	:	0.44047841	,
652	:	0.43293862	,
653	:	0.4250092	,
654	:	0.41703994	,
655	:	0.40880046	,
656	:	0.40021724	,
657	:	0.39181166	,
658	:	0.38326492	,
659	:	0.37444487	,
660	:	0.36518993	,
661	:	0.35540951	,
662	:	0.34614252	,
663	:	0.33687923	,
664	:	0.32745671	,
665	:	0.31755649	,
666	:	0.30757261	,
667	:	0.29772206	,
668	:	0.28785186	,
669	:	0.27806217	,
670	:	0.26796567	,
671	:	0.2577414	,
672	:	0.24832435	,
673	:	0.23936293	,
674	:	0.23047387	,
675	:	0.22197518	,
676	:	0.21419902	,
677	:	0.20695216	,
678	:	0.20021064	,
679	:	0.19377161	,
680	:	0.1875389	,
681	:	0.18153867	,
682	:	0.17597335	,
683	:	0.17057753	,
684	:	0.16497796	,
685	:	0.15952617	,
686	:	0.15447145	,
687	:	0.14972347	,
688	:	0.14514935	,
689	:	0.1409094	,
690	:	0.13669326	,
691	:	0.1316973	,
692	:	0.12610435	,
693	:	0.1206696	,
694	:	0.11565646	,
695	:	0.11113238	,
696	:	0.1068394	,
697	:	0.1024996	,
698	:	0.098283271	,
699	:	0.094152958	,
700	:	0.089880314	,
701	:	0.085422174	,
702	:	0.081255851	,
703	:	0.078449796	,
704	:	0.07630734	,
705	:	0.073793757	,
706	:	0.070399807	,
707	:	0.066398939	,
708	:	0.062552678	,
709	:	0.059116528	,
710	:	0.056308094	,
711	:	0.053846163	,
712	:	0.051289735	,
713	:	0.049146035	,
714	:	0.0467459	,
715	:	0.043786092	,
716	:	0.040513361	,
717	:	0.03749471	,
718	:	0.035530532	,
719	:	0.034330201	,
720	:	0.03333155	,
721	:	0.031907958	,
722	:	0.02953673	,
723	:	0.026947527	,
724	:	0.024187319	,
725	:	0.021385643	,
726	:	0.018943348	,
727	:	0.016925582	,
728	:	0.015922869	,
729	:	0.015040051	,
730	:	0.014096826	,
731	:	0.012969545	,
732	:	0.011366071	,
733	:	0.009767612	,
734	:	0.008291001	,
735	:	0.0069416	,
736	:	0.005667879	,
737	:	0.00434734	,
738	:	0.003187678	,
739	:	0.002056324	,
740	:	0.000949999	,
741	:	1.95E-05	,
742	:	0	,
743	:	0	,
744	:	0	,
745	:	0	,
746	:	0	,
747	:	0	,
748	:	0	,
749	:	0	,
750	:	0	,
751	:	0	,
752	:	0	,
753	:	0	,
754	:	0	,
755	:	0	,
756	:	0	,
757	:	0	,
758	:	0	,
759	:	0	,
760	:	0	,
761	:	0	,
762	:	0	,
763	:	0	,
764	:	0	,
765	:	0	,
766	:	0	,
767	:	0	,
768	:	0	,
769	:	0	,
770	:	0	,
771	:	0	,
772	:	0	,
773	:	0	,
774	:	0	,
775	:	0	,
776	:	0	,
777	:	0	,
778	:	0	,
779	:	0	,
780	:	0	,
781	:	0	,
782	:	0	,
783	:	0	,
784	:	0	,
785	:	0	,
786	:	0	,
787	:	0	,
788	:	0	,
789	:	0	,
790	:	0	,
791	:	0	,
792	:	0	,
793	:	0	,
794	:	0	,
795	:	0	,
796	:	0	,
797	:	0	,
798	:	0	,
799	:	0	,
800	:	0	,
801	:	0	,
802	:	0	,
803	:	0	,
804	:	0	,
805	:	0	,
806	:	0	,
807	:	0	,
808	:	0	,
809	:	0	,
810	:	0	,
811	:	0	,
812	:	0	,
813	:	0	,
814	:	0	,
815	:	0	,
816	:	0	,
817	:	0	,
818	:	0	,
819	:	0	,
820	:	0	,
821	:	0	,
822	:	0	,
823	:	0	,
824	:	0	,
825	:	0	,
826	:	0	,
827	:	0	,
828	:	0	,
829	:	0	,
830	:	0	,
831	:	0	,
832	:	0	,
833	:	0	,
834	:	0	,
835	:	0	,
836	:	0	,
837	:	0	,
838	:	0	,
839	:	0	,
840	:	0	,
841	:	0	,
842	:	0	,
843	:	0	,
844	:	0	,
845	:	0	,
846	:	0	,
847	:	0	,
848	:	0	,
849	:	0	,
850	:	0	,
851	:	0	,
852	:	0	,
853	:	0	,
854	:	0	,
855	:	0	,
856	:	0	,
857	:	0	,
858	:	0	,
859	:	0	,
860	:	0	,
861	:	0	,
862	:	0	,
863	:	0	,
864	:	0	,
865	:	0	,
866	:	0	,
867	:	0	,
868	:	0	,
869	:	0	,
870	:	0	,
871	:	0	,
872	:	0	,
873	:	0	,
874	:	0	,
875	:	0	,
876	:	0	,
877	:	0	,
878	:	0	,
879	:	0	,
880	:	0	,
881	:	0	,
882	:	0	,
883	:	0	,
884	:	0	,
885	:	0	,
886	:	0	,
887	:	0	,
888	:	0	,
889	:	0	,
890	:	0	,
891	:	0	,
892	:	0	,
893	:	0	,
894	:	0	,
895	:	0	,
896	:	0	,
897	:	0	,
898	:	0	,
899	:	0	,
900	:	0	,
901	:	0	,
902	:	0	,
903	:	0	,
904	:	0	,
905	:	0	,
906	:	0	,
907	:	0	,
908	:	0	,
909	:	0	,
910	:	0	,
911	:	0	,
912	:	0	,
913	:	0	,
914	:	0	,
915	:	0	,
916	:	0	,
917	:	0	,
918	:	0	,
919	:	0	,
920	:	0	,
921	:	0	,
922	:	0	,
923	:	0	,
924	:	0	,
925	:	0	,
926	:	0	,
927	:	0	,
928	:	0	,
929	:	0	,
930	:	0	,
931	:	0	,
932	:	0	,
933	:	0	,
934	:	0	,
935	:	0	,
936	:	0	,
937	:	0	,
938	:	0	,
939	:	0	,
940	:	0	,
941	:	0	,
942	:	0	,
943	:	0	,
944	:	0	,
945	:	0	,
946	:	0	,
947	:	0	,
948	:	0	,
949	:	0	,
950	:	0	,
951	:	0	,
952	:	0	,
953	:	0	,
954	:	0	,
955	:	0	,
956	:	0	,
957	:	0	,
958	:	0	,
959	:	0	,
960	:	0	,
961	:	0	,
962	:	0	,
963	:	0	,
964	:	0	,
965	:	0	,
966	:	0	,
967	:	0	,
968	:	0	,
969	:	0	,
970	:	0	,
971	:	0	,
972	:	0	,
973	:	0	,
974	:	0	,
975	:	0	,
976	:	0	,
977	:	0	,
978	:	0	,
979	:	0	,
980	:	0	,
981	:	0	,
982	:	0	,
983	:	0	,
984	:	0	,
985	:	0	,
986	:	0	,
987	:	0	,
988	:	0	,
989	:	0	,
990	:	0	,
991	:	0	,
992	:	0	,
993	:	0	,
994	:	0	,
995	:	0	,
996	:	0	,
997	:	0	,
998	:	0	,
999	:	0	,
1000	:	0	,
1001	:	0	,
1002	:	0	,
1003	:	0	,
1004	:	0	,
1005	:	0	,
1006	:	0	,
1007	:	0	,
1008	:	0	,
1009	:	0	,
1010	:	0	,
1011	:	0	,
1012	:	0	,
1013	:	0	,
1014	:	0	,
1015	:	0	,
1016	:	0	,
1017	:	0	,
1018	:	0	,
1019	:	0	,
1020	:	0	,
1021	:	0	,
1022	:	0	,
1023	:	0	,
1024	:	0	,
1025	:	0	,
1026	:	0	,
1027	:	0	,
1028	:	0	,
1029	:	0	,
1030	:	0	,
1031	:	0	,
1032	:	0	,
1033	:	0	,
1034	:	0	,
1035	:	0	,
1036	:	0	,
1037	:	0	,
1038	:	0	,
1039	:	0	,
1040	:	0	,
1041	:	0	,
1042	:	0	,
1043	:	0	,
1044	:	0	,
1045	:	0	,
1046	:	0	,
1047	:	0	,
1048	:	0	,
1049	:	0	,
1050	:	0	,
1051	:	0	,
1052	:	0	,
1053	:	0	,
1054	:	0	,
1055	:	0	,
1056	:	0	,
1057	:	0	,
1058	:	0	,
1059	:	0	,
1060	:	0	,
1061	:	0	,
1062	:	0	,
1063	:	0	,
1064	:	0	,
1065	:	0	,
1066	:	0	,
1067	:	0	,
1068	:	0	,
1069	:	0	,
1070	:	0	,
1071	:	0	,
1072	:	0	,
1073	:	0	,
1074	:	0	,
1075	:	0	,
1076	:	0	,
1077	:	0	,
1078	:	0	,
1079	:	0	,
1080	:	0	,
1081	:	0	,
1082	:	0	,
1083	:	0	,
1084	:	0	,
1085	:	0	,
1086	:	0	,
1087	:	0	,
1088	:	0	,
1089	:	0	,
1090	:	0	,
1091	:	0	,
1092	:	0	,
1093	:	0	,
1094	:	0	,
1095	:	0	,
1096	:	0	,
1097	:	0	,
1098	:	0	,
1099	:	0	,
1100	:	0	,
1101	:	0	,
1102	:	0	,
1103	:	0	,
1104	:	0	,
1105	:	0	,
1106	:	0	,
1107	:	0	,
1108	:	0	,
1109	:	0	,
1110	:	0	,
1111	:	0	,
1112	:	0	,
1113	:	0	,
1114	:	0	,
1115	:	0	,
1116	:	0	,
1117	:	0	,
1118	:	0	,
1119	:	0	,
1120	:	0	,
1121	:	0	,
1122	:	0	,
1123	:	0	,
1124	:	0	,
1125	:	0	,
1126	:	0	,
1127	:	0	,
1128	:	0	,
1129	:	0	,
1130	:	0	,
1131	:	0	,
1132	:	0	,
1133	:	0	,
1134	:	0	,
1135	:	0	,
1136	:	0	,
1137	:	0	,
1138	:	0	,
1139	:	0	,
1140	:	0	,
1141	:	0	,
1142	:	0	,
1143	:	0	,
1144	:	0	,
1145	:	0	,
1146	:	0	,
1147	:	0	,
1148	:	0	,
1149	:	0	,
1150	:	0	,
1151	:	0	,
1152	:	0	,
1153	:	0	,
1154	:	0	,
1155	:	0	,
1156	:	0	,
1157	:	0	,
1158	:	0	,
1159	:	0	,
1160	:	0	,
1161	:	0	,
1162	:	0	,
1163	:	0	,
1164	:	0	,
1165	:	0	,
1166	:	0	,
1167	:	0	,
1168	:	0	,
1169	:	0	,
1170	:	0	,
1171	:	0	,
1172	:	0	,
1173	:	0	,
1174	:	0	,
1175	:	0	,
1176	:	0	,
1177	:	0	,
1178	:	0	,
1179	:	0	,
1180	:	0	,
1181	:	0	,
1182	:	0	,
1183	:	0	,
1184	:	0	,
1185	:	0	,
1186	:	0	,
1187	:	0	,
1188	:	0	,
1189	:	0	,
1190	:	0	,
1191	:	0	,
1192	:	0	,
1193	:	0	,
1194	:	0	,
1195	:	0	,
1196	:	0	,
1197	:	0	,
1198	:	0	,
1199	:	0	,
1200	:	0
}

**Comment by thomasmansencal** [comment, marked as answer]
Thanks! I got it: It is because you are passing your irradiance source both as the spectral distribution and the illuminant!

```python
SD_LED = colour.SpectralDistribution(data, name=name)

print(colour.colorimetry.sd_to_XYZ_integration(SD_LED, k=1))

W, V = SD_LED.wavelengths, SD_LED.values[..., np.newaxis]

print(np.sum(colour.MSDS_CMFS['CIE 1931 2 Degree Standard Observer'][W] * V, axis=0))
```

```text
[ 82.4402695   82.8543287   78.68796877]
[ 82.4402695   82.8543287   78.68796877]
```

**Reply by heinemannj** [reply]
Jepp - **Your solution is working fine**!
I've tested a lot of options yesterday noon and was more and more confused ...

Possible improvement for documentation:
- Destinguish between the different measurement modes
  Reference: http://www.brucelindbloom.com/Eqn_Spect_to_XYZ.html

Final questions:
**My selfmade spectrometer** (grating and USB-Webcam) connected to optical spectroscopy software "Spectragryph" 
https://www.effemm2.de/index.html
**can only provide relative intensities**.

For transmittance and reflectance measurements this is fully compliant to CIE and ASTM practices - right?
For emmision measurements normally an absolute intensity (power units in Watts or mW) is expected?
Is a caculation of CIE Y from an relative intensity and the further convertion to CIE Lab valid?

Thanks for your patiance and your support!

Cheers
Joerg
